### PR TITLE
Fix Dockerfile in pkg/debug

### PR DESCRIPTION
**Pull request by rucoder**
- FROM and AS must be the same case
- ENV <key> <val> is deprected. ENV key=val must be used

### CODE REVIEW

**Review by OhmSpectator — APPROVED** [review]

**Comment by rene** [comment]
TBH I don't think it's worth this removal, the `vi` available it's the busybox implementation, super limited (it doesn't support even line numbers), for only 7MB, this will force many users to always install vim when running debug container for the first time (after boot), again, super inconvenient... I would put this removal on hold, unless we really can't find any other "less important" package to remove...

**Comment by rucoder** [comment]
> TBH I don't think it's worth this removal, the `vi` available it's the busybox implementation, super limited (it doesn't support even line numbers), for only 7MB, this will force many users to always install vim when running debug container for the first time (after boot), again, super inconvenient... I would put this removal on hold, unless we really can't find any other "less important" package to remove...

@rene I wouldn't bother doing this PR if we have space :) . I cannot submit my PR for TUI application, our current image size is `248.546875` (260620288 bytes). This is the most obvious and easy way to free up a lot of space (>3%) in rootfs

**Comment by rene** [comment]
@rucoder, FWIW these are the sizes of the rootfs of each onboot/service containers:

```
900471d3-e720-4e0e-ba06-6978f635dbb9:~# du -sh /containers/services/*/lower /containers/onboot/*/lower | sort -n | tac
110.9M	/containers/onboot/004-pillar-onboot/lower
101.4M	/containers/services/xen-tools/lower
75.3M	/containers/services/debug/lower
51.4M	/containers/services/vtpm/lower
42.1M	/containers/services/wwan/lower
26.3M	/containers/onboot/002-storage-init/lower
22.3M	/containers/services/edgeview/lower
18.3M	/containers/services/guacd/lower
17.4M	/containers/onboot/005-apparmor/lower
16.8M	/containers/services/newlogd/lower
11.0M	/containers/services/memory-monitor/lower
10.3M	/containers/services/wlan/lower
7.6M	/containers/onboot/003-kdump/lower
7.2M	/containers/onboot/006-measure-config/lower
5.8M	/containers/services/watchdog/lower
900471d3-e720-4e0e-ba06-6978f635dbb9:~# 
```

**Comment by deitch** [comment]
More detail here https://github.com/lf-edge/eve/issues/4322#issuecomment-2399845100

**Comment by rene** [comment]
> More detail here [#4322 (comment)](https://github.com/lf-edge/eve/issues/4322#issuecomment-2399845100)

Thanks @deitch , I've missed your much more deep analysis on the sizes.... in anycase, @rucoder , how much space do you want to release for the TUI? 20MB, 50MB would be enough? I don't see any problems in increase the maximum image size to, let's say, 280MB, 290MB... it doesn't invalidate the concerns, but at least we'll have more time to look closer on what can be effectively done to reduce the image size...

**Comment by rucoder** [comment]
> > More detail here [#4322 (comment)](https://github.com/lf-edge/eve/issues/4322#issuecomment-2399845100)
> 
> Thanks @deitch , I've missed your much more deep analysis on the sizes.... in anycase, @rucoder , how much space do you want to release for the TUI? 20MB, 50MB would be enough? I don't see any problems in increase the maximum image size to, let's say, 280MB, 290MB... it doesn't invalidate the concerns, but at least we'll have more time to look closer on what can be effectively done to reduce the image size...

PR https://github.com/lf-edge/eve/pull/4329 submitted. I tried to avoid it to reduce any potential risk, but let's increase it.

**Comment by rene** [comment]
> > > More detail here [#4322 (comment)](https://github.com/lf-edge/eve/issues/4322#issuecomment-2399845100)
> > 
> > 
> > Thanks @deitch , I've missed your much more deep analysis on the sizes.... in anycase, @rucoder , how much space do you want to release for the TUI? 20MB, 50MB would be enough? I don't see any problems in increase the maximum image size to, let's say, 280MB, 290MB... it doesn't invalidate the concerns, but at least we'll have more time to look closer on what can be effectively done to reduce the image size...
> 
> PR #4329 submitted. I tried to avoid it to reduce any potential risk, but let's increase it. I mean the size :) not, risk

Thanks @rucoder , it makes sense, also if we are adding more features, specially graphical stuff, that's expected to have an increase in the image size, so I think we are still safe with a 350MB limit.... can we close this PR?

**Comment by rucoder** [comment]
> > > > More detail here [#4322 (comment)](https://github.com/lf-edge/eve/issues/4322#issuecomment-2399845100)
> > > 
> > > 
> > > Thanks @deitch , I've missed your much more deep analysis on the sizes.... in anycase, @rucoder , how much space do you want to release for the TUI? 20MB, 50MB would be enough? I don't see any problems in increase the maximum image size to, let's say, 280MB, 290MB... it doesn't invalidate the concerns, but at least we'll have more time to look closer on what can be effectively done to reduce the image size...
> > 
> > 
> > PR #4329 submitted. I tried to avoid it to reduce any potential risk, but let's increase it. I mean the size :) not, risk
> 
> Thanks @rucoder , it makes sense, also if we are adding more features, specially graphical stuff, that's expected to have an increase in the image size, so I think we are still safe with a 350MB limit.... can we close this PR?

@rene let me remove the second commit. the one for Dockerfile is still useful

**Comment by rucoder** [comment]
@rene I repurposed the PR.

**Review by christoph-zededa — APPROVED** [review]

**Review by OhmSpectator — APPROVED** [review]
Like

**Comment by rene** [comment]
> @rene I repurposed the PR.

@rucoder , fair enough... LGTM...

**Review by rene — APPROVED** [review]

**Comment by eriknordmark** [comment]
> > > TBH I don't think it's worth this removal, the `vi` available it's the busybox implementation, super limited (it doesn't support even line numbers), for only 7MB, this will force many users to always install vim when running debug container for the first time (after boot), again, super inconvenient... I would put this removal on hold, unless we really can't find any other "less important" package to remove...
> > 
> > 
> > @rene I wouldn't bother doing this PR if we have space :) . I cannot submit my PR for TUI application, our current image size is `248.546875` (260620288 bytes). This is the most obvious and easy way to free up a lot of space (>3%) in rootfs
> 
> The IMG partition size is 512MB, the 250MB it's a soft limit, why don't you want to increase it?

@rene Correct that 250MB is a soft limit, but the hard limit is 300MB for the 99% of deployed devices (or maybe 100%). 512MB is only those which have been installed from scratch from master over the last 6? months.

File: pkg/debug/Dockerfile [hunk 1]
```diff
@@ -8,34 +8,34 @@
...
-ENV BUILD_PKGS abuild curl tar make linux-headers patch g++ git gcc gpg ncurses-dev autoconf openssl-dev zlib-dev
+FROM lfedge/eve-alpine:82df60e43ab9f8c935584b8c7b4d0a4b0271d608 AS build
+ENV BUILD_PKGS="abuild curl tar make linux-headers patch g++ git gcc gpg ncurses-dev autoconf openssl-dev zlib-dev"
 # Feel free to add additional packages here, but be aware that
 # EVE's rootfs image can be no larger than 300Mb (and don't
 # forget to check on all supported architectures: e.g. arm64
 # binaries are typically larger and amd64 ones).
 # RUN apk add --no-cache gdb valgrind
-ENV PKGS openssl tini util-linux ca-certificates pciutils usbutils vim tcpdump perf strace iproute2-minimal curl zlib
+ENV PKGS="openssl tini util-linux ca-certificates pciutils usbutils vim tcpdump perf strace iproute2-minimal curl zlib"
```
Comment (eriknordmark) [inline pkg/debug/Dockerfile:20]: Which package(s) did you remove? I see no difference.

Comment (OhmSpectator) [reply pkg/debug/Dockerfile:20]: > Which package(s) did you remove? I see no difference.

Now, this PR is only about fixing syntax in the Docker file)
https://github.com/lf-edge/eve/pull/4328#issuecomment-2401815105

**Review by OhmSpectator — APPROVED** [review]
run the tests?

**Pull request merged**